### PR TITLE
REACT HW 5

### PR DESCRIPTION
**Pull request by ThisIsAntony**
# React task 5

## Summary
Based on your previous tasks, add 3 routes to your application: login page(/login), your profile page(/pofile), and your page with cards(/cards).
 
### Requirements
Your app should redirect the user to a login page (/login). You can do it using the history API from react-router. On this login page create a form with 2 inputs (username and password) with a submit button (use mocked logic for login).
If the user successfully logged in, he can be able to view the profile(/profile) page and cards(/cards) page
Profile page contains any data you like. (Add 2-3 sentences about yourself)
Cards page should contain your <CardsContainer /> component with all the logic you’ve done before (no additional code required)

Important note: if the user in not logged in he should NOT able to view /profile and /cards pages

### CODE REVIEW

File: src/components/main/main.jsx [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
-import styles from "./main.module.scss";
-
-const Main = ({ avatar }) => {
-  return (
-    <main className={styles.content}>
-      <img src={avatar.image} alt={avatar.alt} className={styles.image} />
-      <CardsContainer />
-    </main>
-  );
+import { useState } from 'react';
```
Comment (kaseopea) [inline src/components/main/main.jsx:1]: Unused imports

File: src/components/login/login-container.jsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+			...loginData,
+			[fieldName]: value
+		});
+		if (value) return setIsFill({ ...isFill, [fieldName]: true });
+		setIsFill({ ...isFill, [fieldName]: false });
+	};
+
+	const handleSubmit = (e) => {
+		e.preventDefault();
+		const isAllFiedlFill = Object.values(loginData).every((item) => item);
```
Comment (kaseopea) [inline src/components/login/login-container.jsx:37]: I assume you meant `isAllFieldsFilled`?)

File: src/components/login/login-container.jsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import { useHistory } from 'react-router';
+import Login from './login';
+
+const LoginContainer = ({ setIsAuth }) => {
+	const history = useHistory();
+	const [ loginData, setLoginData ] = useState({
+		login: '',
+		password: ''
+	});
+	const [ isFill, setIsFill ] = useState({
```
Comment (kaseopea) [inline src/components/login/login-container.jsx:11]: According DRY principle it's better to create additional variable for:
```
{ login: false, passwords: false }

```
For example `credentialsDefaults` - so you would have much cleaner code.

Comment (ThisIsAntony) [reply src/components/login/login-container.jsx:18]: done

File: src/components/login/login-container.jsx (hunk 2 above)
Comment (ThisIsAntony) [reply src/components/login/login-container.jsx]: done

File: src/components/main/main.jsx (hunk 1 above)
Comment (ThisIsAntony) [reply src/components/main/main.jsx]: done

**Pull request merged**